Posh-ACME/Plugins/Infomaniak.ps1

```diff
@@ -0,0 +1,262 @@
+function Get-CurrentPluginType { 'dns-01' }
+
+Function Add-DnsTxt {
+    [CmdletBinding(DefaultParameterSetName='Secure')]
+    param(
+        [Parameter(Mandatory,Position = 0)]
+        [string]$RecordName,
+        [Parameter(Mandatory,Position = 1)]
+        [string]$TxtValue,
+        [Parameter(ParameterSetName='Secure',Mandatory,Position=2)]
+        [securestring]$InfomaniakToken,
+        [Parameter(ParameterSetName='Insecure',Mandatory,Position=2)]
+        [string]$InfomaniakTokenInsecure,
+        [Parameter(ValueFromRemainingArguments)]
+        $ExtraParams
+    )
+
+    $apiRoot = 'https://api.infomaniak.com'
+
+    # un-secure the password so we can add it to the auth header
+    if ('Secure' -eq $PSCmdlet.ParameterSetName) {
+        $InfomaniakTokenInsecure = (New-Object PSCredential "user",$InfomaniakToken).GetNetworkCredential().Password
+    }
+    $restParams = @{
+        Headers = @{
+            Authorization = "Bearer $InfomaniakTokenInsecure"
+            Accept = 'application/json'
+        }
+        ContentType = 'application/json'
+    }
+
+    # find matching ZoneID to check, if the records exists already
+    if (-not ($zone = Find-InfomaniakZone $RecordName $restParams)) {
+        throw "Unable to find matching zone for $RecordName"
+    }
+
+    # separate the portion of the name that doesn't contain the zone name
+    $recShort = ($RecordName -ireplace [regex]::Escape($zone.name), [string]::Empty).TrimEnd('.')
+
+    # Get a list of existing TXT records for this record name
+    try {
+        Write-Verbose "Searching for existing TXT record"
+        $recs = Invoke-RestMethod "$apiRoot/1/domain/$($zone.id)/dns/record" `
+            @restParams @Script:UseBasic -EA Stop
+    } catch { throw }
+
+    # check for a matching record
+    $rec = $recs.data | Where-Object {
+        $_.type -eq 'TXT' -and
+        $_.source -eq $recShort -and
+        $_.target -eq $TxtValue
+    }
+
+    if ($rec) {
+        Write-Debug "Record $RecordName already contains $TxtValue. Nothing to do."
+    } else {
+        # create request body schema
+        $body = @{
+            type = 'TXT'
+            source = $recShort
+            target = $TxtValue
+            ttl = 600
+        }
+        $json = $body | ConvertTo-Json
+
+        try {
+            $response = Invoke-RestMethod "$apiRoot/1/domain/$($zone.id)/dns/record" -Method Post -Body $json `
+                @restParams @Script:UseBasic -EA Stop
+            if($response.result -eq 'success')
+            {
+                Write-Verbose "Record $RecordName added with value $TxtValue."
+            }
+            {
+                throw "Record $RecordName with value $TxtValue could not be added."
+            }
```

Comment:
I think you're missing an `else` in here.

Reply:
yes. Strange that during Publish-Challenge PS didn't throw an error.